packages/@sanity/default-layout/src/navbar/search/utils/isRecentSearchTerms.ts

```diff
@@ -0,0 +1,12 @@
+import type {SearchTerms} from '@sanity/base'
+import type {RecentSearchTerms} from '../datastores/recentSearches'
+
+export function isRecentSearchTerms(
+  terms: SearchTerms | RecentSearchTerms
+): terms is RecentSearchTerms {
+  const recentSearchTerms = terms as RecentSearchTerms
+  return (
+    typeof recentSearchTerms.__recentTimestamp !== 'undefined' &&
```

Comment:
Another datapoint that wrapper object could be in order; this is another callsite that must change if RecentSearchTerms gets more __ props